### PR TITLE
Add displs class to manage data layout

### PR DESCRIPTION
**Pull request by uphoffc**
Useful class to handle data with varying sizes.
E.g.
```
std::vector<int> count{2,2,4,4,4};
Displs<int> layout(count);
for (auto [i,j] : layout) {
    recvcounts[i] += sharedRanksRecvCount[j];
}
```

### CODE REVIEW

File: include/mneme/displs.hpp [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+template <typename IntT> class Displs {
+public:
+    Displs() : displs(1, 0) {}
+
+    Displs(std::vector<IntT> const& count) { make(count); }
+
+    void make(std::vector<IntT> const& count) {
+        displs.resize(count.size() + 1);
+        displs[0] = 0;
+        for (IntT p = 0; p < count.size(); ++p) {
```
Comment (krenzland) [inline include/mneme/displs.hpp]: This is going to give a warning if IntT is signed - so we better cast count.size to the proper type?

File: include/mneme/displs.hpp [hunk 2]
```diff
@@ -0,0 +1,94 @@
...
+        next();
+        return *this;
+    }
+    DisplsIterator operator++(int) {
+        DisplsIterator copy(*this);
+        ++(*this);
+        return copy;
+    }
+
+    value_type operator*() { return std::make_pair(p, i); }
```
Comment (krenzland) [inline include/mneme/displs.hpp]: Just a hint: You could also just  return {p, i}

File: include/mneme/displs.hpp [hunk 3]
```diff
@@ -0,0 +1,94 @@
...
+        } else {
+            i = displs[0];
+            next();
+        }
+    }
+
+    bool operator!=(DisplsIterator const& other) {
+        return i != other.i || p != other.p || &displs != &other.displs;
+    }
+    bool operator==(DisplsIterator const& other) { return !(*this != other); }
```
Comment (krenzland) [inline include/mneme/displs.hpp]: Is there a reason why this isnt *this == other?

File: include/mneme/displs.hpp [hunk 4]
```diff
@@ -0,0 +1,94 @@
...
+            ++p;
+            i = displs[p];
+        }
+    }
+
+    std::vector<IntT> const& displs;
+    std::size_t p, i;
+};
+
+template <typename IntT> class Displs {
```
Comment (krenzland) [inline include/mneme/displs.hpp]: Can we write the full name of the class? At a first glance it was not obvious what Displs stands for

**Review by krenzland — CHANGES_REQUESTED** [review]

File: include/mneme/displs.hpp (hunk 3 above)
Comment (uphoffc) [reply include/mneme/displs.hpp]: ?

**Review by krenzland — APPROVED** [review]

**Pull request merged**